### PR TITLE
include the test files in the distribution tar, fixes #257

### LINKED ISSUE #257: missing testsuite files in tarball
When running the testsuite for the Stackage Nightly build:

```
    /var/stackage/work/unpack-dir/unpacked/haskell-awk-1.2-65b8317d0e3e0a1c81e42c09b5b0cad3c4729efba89be675cc344d3608c6cf8c/src/System/Console/Hawk/UserPrelude.hs:24: failure in expression `testC "default"'
    expected: {-# LANGUAGE ExtendedDefaultRules, OverloadedStrings #-}
              module System.Console.Hawk.CachedPrelude where
              {-# LINE 2 "tests/preludes/default/prelude.hs" #-}
              import Prelude
              import qualified Data.ByteString.Lazy.Char8 as B
              import qualified Data.List as L
     but got: *** Exception: tests/preludes/default/prelude.hs: openBinaryFile: does not exist (No such file or directory)
              ^
    
    Examples: 173  Tried: 172  Errors: 0  Failures: 1
```
Should all the tests/**.hs files be added to the .cabal file?

### PR DESCRIPTION
**Pull request by gelisam**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -117,4 +117,11 @@ jobs:
...
       run: |
-        cabal v2-test
+        rm -rf dist-newstyle/sdist sdist
+        cabal sdist
+        cp -r dist-newstyle/sdist .
+        cd sdist
+        tar xzvf *.tar.gz
+        rm -rf *.tar.gz
+        cd *
+        cabal test
```
Comment (gelisam) [inline .github/workflows/ci.yml:127]: this dance seems to test the right thing, but seems overly-convoluted. is there a simpler way?

File: haskell-awk.cabal [hunk 2]
```diff
@@ -14,13 +14,31 @@ category:       Console
 homepage:       https://github.com/gelisam/hawk#readme
 bug-reports:    https://github.com/gelisam/hawk/issues
 author:         Mario Pastorelli <pastorelli.mario@gmail.com>,  Samuel Gélineau <gelisam@gmail.com>
-maintainer:     Mario Pastorelli <pastorelli.mario@gmail.com>,  Samuel Gélineau <gelisam@gmail.com>
+maintainer:     Samuel Gélineau <gelisam@gmail.com>, Jens Petersen <juhpetersen@gmail.com>
```
Comment (gelisam) [inline haskell-awk.cabal:17]: looks like we forgot to regenerate the `.cabal` file after modifying the `package.yaml` file last time. this means that when we released haskell-awk-1.2, we released without this change. I have updated it via a hackage revision.

**Comment by gelisam** [comment]
tests are fixed. @juhp, what's the process for re-enabling the tests on the stackage side? Do I just open a PR reverting https://github.com/commercialhaskell/stackage/commit/4042c4c97c15cc31a9573089840191359ae7c504?

**Comment by gelisam** [comment]
oh wait, we probably need to release a new version which includes the fix, otherwise stackage will still test haskell-awk-1.2, won't they?

**Comment by juhp** [comment]
Ah yes that's true, thanks

**Pull request merged**